### ISSUE TITLE
How to resize canvas and center image without resizing?

### ISSUE THREAD

**Issue by joerg-walter-de** [lovell/sharp#2881]
My image is 100x100 pixel. The resulting image must be 200x200 pixel. The original image must be centered so that there is a 50 pixel border around it. How do I do that?

**Comment by lovell** [comment]
Hi, did you see https://sharp.pixelplumbing.com/api-resize#extend ?

**Comment by joerg-walter-de** [comment]
@lovell Yes, I saw that. Thanks. I find the interface a bit awkward, because for achieving what I want I would have to calculate the extension values first. With the resize interface I get the high level center-option. 

I can think of a canvas-size function canvas(width, height, positioning) -> canvas(200, 200, 'center') to make that comfortable.

**Comment by joerg-walter-de** [comment]
I came up with this one

```
sharp({
                create: {
                    width: AVATAR_IMAGE_MAX_HEIGHT, 
                    height: AVATAR_IMAGE_MAX_WIDTH,
                    channels: 4,
                    background: { r: 0, g: 0, b: 0, alpha: 0.0 }
                }
            })
                .png()
                .composite([{ input: avatarImageBuffer, gravity: 'center' }])
                .toBuffer()
            );
```

Not sure if this is efficient or not.

**Comment by lovell** [comment]
Yes, your suggestion will work, and I think more closely matches the model you've described.

There's a slight CPU cost due to the use of RGBA as this will require a premultiply/unpremultiply round-trip when compositing, but I think for your use case this is still the best approach.

**Comment by lovell** [comment]
I hope this information helped. Please feel free to re-open with more details if further assistance is required.
